### PR TITLE
Email template post

### PR DESCRIPTION
**Pull request by savyoncohen**
@amitaibu

### CODE REVIEW

**Comment by amitaibu** [comment]
Great start!

File: app/_posts/2016-07-31-email-template-on-drupal.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+For our website we have all kinds of browsers testing tools, for an email their is a nice tool which is called <a href="https://mailchimp.com/">Mailchimp</a>. In Mailchimp you can build any email template you want, with a nice wysiwyg editor. you can add and design almost anything you want : linked buttons, social networks linked icons, images and videos.
+Here is an example for an email template I created with the Mailchimp editor, without writing even one single line of HTML or CSS at all:
+
+{% include thumbnail.html  image_path="assets/images/posts/email-template/.jpg" caption="Post has a happy end - we were able to migrate files and attach to a node!" %}
+
+Behind the scenes, Mailchimp have converted my design into nested tables with the mostly mail supported CSS rules. There is also an option to see the source on the Mailchimp editor, and i can change anything i want.
```
Comment (amitaibu) [inline app/_posts/2016-07-31-email-template-on-drupal.md]: replace `i` with `I` here and below.

**Comment by RachelBaram** [comment]
@savyoncohen Let me know when you are ready for me to proofread your blog. ;)

**Comment by savyoncohen** [comment]
Hi @RachelBaram , 
can you please review my post?
thanks.

**Comment by RachelBaram** [comment]
There is some awkward phrasing in the technical sections that will require your extra attention @amitaibu.

File: app/_posts/2016-07-31-email-template-on-drupal.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+title: "Creating a Dynamic Email Template in Drupal"
+tags:
+  - Drupal-planet
+  - Tutorial
+  - Email
+permalink: "/content/dynamic-email-template"
+layout: post
+author: Savyoncohen
+<!-- image: /assets/images/posts/elm-electron-build/thumb.jpg -->
+published: false
```
Comment (amitaibu) [inline app/_posts/2016-07-31-email-template-on-drupal.md]: remove

File: app/_posts/2016-07-31-email-template-on-drupal.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
+---
+title: "Creating a Dynamic Email Template in Drupal"
+tags:
+  - Drupal-planet
+  - Tutorial
+  - Email
+permalink: "/content/dynamic-email-template"
+layout: post
+author: Savyoncohen
+<!-- image: /assets/images/posts/elm-electron-build/thumb.jpg -->
```
Comment (amitaibu) [inline app/_posts/2016-07-31-email-template-on-drupal.md]: we need a thumb

File: app/_posts/2016-07-31-email-template-on-drupal.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
+---
+title: "Creating a Dynamic Email Template in Drupal"
+tags:
+  - Drupal-planet
+  - Tutorial
```
Comment (amitaibu) [inline app/_posts/2016-07-31-email-template-on-drupal.md]: remove

File: app/_posts/2016-07-31-email-template-on-drupal.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  - Email
```
Comment (amitaibu) [inline app/_posts/2016-07-31-email-template-on-drupal.md]: remove

File: app/_posts/2016-07-31-email-template-on-drupal.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+permalink: "/content/dynamic-email-template"
+layout: post
+author: Savyoncohen
```
Comment (amitaibu) [inline app/_posts/2016-07-31-email-template-on-drupal.md]: this doesn't appear right - update the config.yml please

gizra_s_blog

File: app/_posts/2016-07-31-email-template-on-drupal.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+By doing this, we can focus on our logic without worrying about how it may look with different dynamic content. Focusing on Frontend and Backend as seperate tasks, makes building websites easier and faster. Even fixing bugs discovered while implementing dynamic content can now be easily fixed. Our
+<a href="http://www.gizra.com/content/custom-css-as-contrib-with-jekyll/">No more CSS in your Drupal Theme!<a> blog post talks more extensivly about the way we work with Drupal themes.
+The same approach is implemented when we create an email template. We first build the email markup with static content, and then we use it to create dynamic content messages. Let's get started:
+
+## 1) Creating an email template
```
Comment (amitaibu) [inline app/_posts/2016-07-31-email-template-on-drupal.md]: remove the 1), 2) from the headers

File: app/_posts/2016-07-31-email-template-on-drupal.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+      'template' => 'email-content-template',
+      'path' => drupal_get_path('module', 'my_module') . '/templates',
+    ),
+```
+
+Now lets use this theme to create the message. In this example, the variables are the user name on the wrapper template and articles on my content template.
+I created a message type named newsletter and I'm using the message module to send my message:
+
+```php
+function my_module_send_messages_immediate() {
```
Comment (amitaibu) [inline app/_posts/2016-07-31-email-template-on-drupal.md]: do we need all this code to understand the concept? Maybe link to a repo with this module?

Also indentation is wrong.

**Comment by savyoncohen** [comment]
@amitaibu 
I made some changes and added a repository.

File: app/_posts/2016-09-12-email-template-in-drupal.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+We don't use custom CSS in Drupal themes. When we start building a website, we divide our work into several milestones, the first is creating a clean and static markup, using Jekyll. At Gizra, we take pixel perfect very seriously, and by doing the markup first, we can concentrate on building our App pages exactly the way they are suppose to look, test their responsiveness, show our clients a first draft, and fix bugs before getting into the logic. We use gulp to compile the SASS files into one CSS file, and after doing that, we copy the CSS file to the themes folder. Then we take our static pages, cut them into pieces, and use them in Drupal themes and plugins.
+By doing this, we can focus on our logic without worrying about how it may look with different dynamic content. Focusing on Frontend and Backend as seperate tasks, makes building websites easier and faster. Even fixing bugs discovered while implementing dynamic content can now be easily fixed. Our
+<a href="http://www.gizra.com/content/custom-css-as-contrib-with-jekyll/">No more CSS in your Drupal Theme!</a> blog post talks more extensivly about the way we work with Drupal themes.
```
Comment (amitaibu) [inline app/_posts/2016-09-12-email-template-in-drupal.md]: use `[]()` syntax instead of the `<a href>

**Comment by savyoncohen** [comment]
@amitaibu 
I changed the text on the second step and changed the links to md syntax.

**Comment by amitaibu** [comment]
Update: There are a few images that are not used, I'll remove them. Also, next time please run a spell checker.

**Comment by amitaibu** [comment]
I have changed the last section to be less verbose and just give hints on the direction, since it was hard to follow and the code linked didn't help much.

I think it's now approachable by devs. @RachelBaram can you please give it a last proofread.

File: app/_posts/2016-09-12-email-template-in-drupal.md [hunk 10]
```diff
@@ -2,6 +2,7 @@
 title: "Creating a Dynamic Email Template in Drupal"
 tags:
   - Drupal-planet
+  - Tutorial
```
Comment (RachelBaram) [inline app/_posts/2016-09-12-email-template-in-drupal.md]: As part of the tag project I am running, I am interested in starting (and retroactively) to use the 'tutorial' tag.

File: app/_posts/2016-09-12-email-template-in-drupal.md [hunk 11]
```diff
@@ -28,34 +29,34 @@ The same approach is implemented when we create an email template. We first buil
...
 
 
-Our way to overcome this problem is to design our emails with nested tables since they are supported by most email services. Using tables is still not enough. To make sure that our email will look the way we want, we need to set a specific width for each table cell and that's a lot of work. After creating our email template we need to find a way to test it and make sure that it looks the way we meant on every media or mail service.
+Our way to overcome this problem is to design our emails with nested tables since they are supported by most email services. However, this is still not enough. To make sure that our email will look the way we want, we need to set a specific width for each table cell and that's a lot of work. After creating our email template we need to find a way to test it and make sure that it looks the way we meant on every media or mail service.
```
Comment (RachelBaram) [inline app/_posts/2016-09-12-email-template-in-drupal.md]: "and that's a lot of work."  It is not clear if this is said to show a solution that is easier or just stating that this part will take the developer some time to do.

File: app/_posts/2016-09-12-email-template-in-drupal.md [hunk 12]
```diff
@@ -28,34 +29,34 @@ The same approach is implemented when we create an email template. We first buil
...
+I upload my images to Mailchimp's cloud, so I won't have to worry about my users email software blocking images attached to the email.
 
 Mailchimp also gives us the opportunity to test our email template on desktop, mobile, and inbox software such as different versions of Outlook, Gmail, and Yahoo on different browsers.
-After finishing the email template, I can go to my templates list and export it as an HTML file, which is combined with the inline CSS. Next we need to wire it into Drupal.
+After finishing the email template, I go to my templates list and export it as an HTML file, which is combined with the inline CSS. Next we need to wire it into Drupal.
 
 ## From Static to Dynamic
 
-The idea is now to take the static HTML and slowly convert it to by dynamic.
+The idea is now to take the static HTML and slowly convert it to dynamic.
```
Comment (RachelBaram) [inline app/_posts/2016-09-12-email-template-in-drupal.md]: I was not sure what "by dynamic" is meant to say.  I would write "dynamic HTML" if that is indeed what you meant.

File: app/_posts/2016-09-12-email-template-in-drupal.md [hunk 13]
```diff
@@ -28,34 +29,34 @@ The same approach is implemented when we create an email template. We first buil
...
 
-Instead of calling `drupl_mail()` directly we usually use the Message and Message notify modules to send the emails. It has the advantage that each mail is saved in the system, so it's easier to track them (and find bugs, if there were any).
+Instead of calling `drupal_mail()` directly we usually use the Message and Message notify modules to send the emails. It has the advantage that each mail is saved in the system, so it's easier to track (and find bugs, if there are any).
```
Comment (RachelBaram) [inline app/_posts/2016-09-12-email-template-in-drupal.md]: Please check to make sure this is a real typo.

**Comment by RachelBaram** [comment]
Did another round of text edits with a couple of questions/comments. Cheers.

File: app/_posts/2016-09-12-email-template-in-drupal.md (hunk 10 above)
Comment (amitaibu) [inline app/_posts/2016-09-12-email-template-in-drupal.md]: Sounds ok, but in this specific case, It's not a tutorial

File: app/_posts/2016-09-12-email-template-in-drupal.md (hunk 12 above)
Comment (amitaibu) [inline app/_posts/2016-09-12-email-template-in-drupal.md]: yeah, it should be `to be dynamic`

File: app/_posts/2016-09-12-email-template-in-drupal.md (hunk 10 above)
Comment (amitaibu) [inline app/_posts/2016-09-12-email-template-in-drupal.md]: (I'll change)

File: app/_posts/2016-09-12-email-template-in-drupal.md (hunk 12 above)
Comment (amitaibu) [inline app/_posts/2016-09-12-email-template-in-drupal.md]: sorry, It looks ok now.

Comment (amitaibu) [inline app/_posts/2016-09-12-email-template-in-drupal.md]: I've changed to `... HTML and slowly introduce the dynamic content.`

**Comment by amitaibu** [comment]
Merged and deployed - thanks

http://www.gizra.com/content/dynamic-email-template/

**Comment by amitaibu** [comment]
This PR broke your image on the team page :/

**Comment by amitaibu** [comment]
Pushed a quick fix, until #498 is in.
